AK/Optional.h

```diff
@@ -629,6 +632,110 @@ requires(IsLvalueReference<T>) class [[nodiscard]] Optional<T> {
     RemoveReference<T>* m_pointer { nullptr };
 };
 
+template<typename T>
+requires(requires { Traits<T>::special_optional_empty_value; })
+class [[nodiscard]] Optional<T> : public OptionalBase<T> {
+
+    // Note: We need to go through this helper to not materialize a temporary at compile time
+    //       which needs to be destroyed at compile time.
+    ALWAYS_INLINE constexpr static auto the_empty_value() -> T
+    {
+        if constexpr (IsSame<decltype(Traits<T>::special_optional_empty_value), T>)
+            return Traits<T>::special_optional_empty_value;
+        else if constexpr (IsCallableWithArguments<decltype(Traits<T>::special_optional_empty_value), T>)
+            return Traits<T>::special_optional_empty_value();
```

Comment:
Is this branch ever used? We seem to always define `special_optional_empty_value(Badge<Optional<T>>)`

Reply:
Yes, not using this version yet, as all special values are currently "illegal" values, so all APIs are currently `Badge`d
left it in as it might come usefull

Note the immediate value branch is also currently unused